core/src/main/java/com/linecorp/armeria/common/AggregatedHttpRequest.java

```diff
@@ -298,4 +298,23 @@ default Locale selectLocale(Locale... supportedLocales) {
     default HttpRequest toHttpRequest() {
         return HttpRequest.of(headers(), content(), trailers());
     }
+
+    /**
+     * Converts this request into a new complete {@link HttpRequest} replacing the {@link #headers()}
+     * with the specified {@link RequestHeaders}.
+     *
+     * @return the new {@link HttpRequest} converted from this request.
+     */
+    default HttpRequest toHttpRequest(RequestHeaders headers) {
+        requireNonNull(headers, "headers");
+        final HttpData content = content();
+        final RequestHeadersBuilder builder = headers.toBuilder();
```

Comment:
Should we use `headers` as is if it has a valid content-length already?

Reply:
That's a good suggestion. 👍